core/kazoo_number_manager/src/knm_port_request.erl

```diff
@@ -563,6 +594,68 @@ transition_numbers(PortReq) ->
             end
     end.
 
+update_used_by(_Numbers, 'undefined') -> ok;
+update_used_by(Numbers, App) ->
+    lager:debug("update number ~p with used_by ~p", [Numbers, App]),
+    knm_numbers:assign_to_app(Numbers, App).
+
+-spec app_used_by_portin(kz_term:ne_binary(), kz_json:object()) -> 'undefined' | kz_json:object().
+app_used_by_portin(Number, JObj) ->
+    Numbers = kzd_port_requests:numbers(JObj),
+    kz_json:get_value([Number, ?USED_BY_KEY], Numbers).
+
+maybe_reconcile_app_used_by(Num, AccountId, JObj) ->
+    AccountDb = kz_util:format_account_db(AccountId),
+    PortInApp = app_used_by_portin(Num, JObj),
+    AppUsage = used_by_which_app(AccountDb, Num),
+    maybe_reconcile_app_used_by(PortInApp, AppUsage, Num, JObj).
+
+maybe_reconcile_app_used_by(App, App, _Num, _JObj) ->
+    {'ok', App};
+maybe_reconcile_app_used_by(_App, App, Num, JObj) ->
+    lager:warning("port in number ~p app ~p replaced by correct app ~p", [Num, _App, App]),
+    _Ok = assign_to_app(Num, App, JObj),
+    {'ok', App}.
+
+query_app_view(AccountDb, View, Num) ->
+    Options = [{'key', Num}],
+    Result = kz_datamgr:get_results(AccountDb, View, Options),
+    case Result of
+        {'ok', []} ->
+            lager:debug("not found number ~p via view ~p", [Num, View]),
+            {error, not_found};
+        {'ok', Doc} ->
+            {'ok', Doc};
+        E ->
+            lager:debug("Error find number ~p document view ~p error ~p", [Num, View, E]),
+            {error, E}
+    end.
+
+is_used_by_app(AccountDb, Num, App) ->
+    View = maps:get(App, ?APP_VIEW_MAP, undefined),
```

Comment:
you don;t actually need this. Just pattern match in function header for callflow and trunk store